src/lib.rs

```diff
@@ -353,9 +353,19 @@ impl<'a> Extract<'a> {
                 let mut archive = zip::ZipArchive::new(source)?;
                 for i in 0..archive.len() {
                     let mut file = archive.by_index(i)?;
-                    let path = into_dir.join(file.name());
-                    let mut output = fs::File::create(path)?;
-                    io::copy(&mut file, &mut output)?;
+                    let outpath = into_dir.join(file.sanitized_name());
+
+                    if (&*file.name()).ends_with('/') {
```

Comment:
can you use `outpath.is_dir()` instead to be windows compatible

Reply:
That was copy pasted from zip-rs example, but yes I can.

Reply:
Doing that causes the program to think everything is a file. I think this is mostly due to zip-rs handling folders as files and only making the difference using the ending /.

Reply:
hm, that's unfortunate... Do you know if that's also the case on windows? Could you maybe setup a test zip file with nested dirs and add windows-cfg'd/linux-cfg'd tests that check this?

Reply:
I think the reason for the is_dir to not work is because of the sanitized_name call, which might be removing the trailing slash.

I'm a bit(very) short on time currently, so I don't want to spend a lot of time on this bug fix writing unit tests. Sorry!